pinot-controller/src/main/java/org/apache/pinot/controller/recommender/rules/io/params/RecommenderConstants.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.controller.recommender.rules.io.params;
+
+public class RecommenderConstants {
+  public static class InvertedSortedIndexJointRule {
+    public static final double DEFAULT_PERCENT_SELECT_FOR_FUNCTION = 0.5d;
+    public static final double DEFAULT_PERCENT_SELECT_FOR_TEXT_MATCH = 0.5d;
+    public static final double DEAFULT_PERCENT_SELECT_FOR_RANGE = 0.5d;
+    public static final double DEAFULT_PERCENT_SELECT_FOR_REGEX = 0.5d;
+    public static final double DEFAULT_PERCENT_SELECT_FOR_ISNULL = 0.5d;
+    public static final double DEFAULT_THRESHOLD_MIN_AND_PREDICATE_INCREMENTAL_VOTE = 0.6d;
+    public static final double DEFAULT_THRESHOLD_RATIO_MIN_AND_PREDICATE_TOP_CANDIDATES = 0.8d;
+    public static final double DEFAULT_THRESHOLD_RATIO_MIN_GAIN_DIFF_BETWEEN_ITERATION = 0.05d;
+    public static final int DEFAULT_MAX_NUM_ITERATION_WITHOUT_GAIN = 3;
+    public static final double DEFAULT_THRESHOLD_RATIO_MIN_NESI_FOR_TOP_CANDIDATES = 0.7d;
+  }
+
+  public static class RulesToExecute {
+    public static final boolean DEFAULT_RECOMMEND_FLAG_QUERY = true;
+    public static final boolean DEFAULT_RECOMMEND_VARIED_LENGTH_DICTIONARY = true;
+    public static final boolean DEFAULT_RECOMMEND_KAFKA_PARTITION = true;
+    public static final boolean DEFAULT_RECOMMEND_PINOT_TABLE_PARTITION = true;
+    public static final boolean DEFAULT_RECOMMEND_INVERTED_SORTED_INDEX_JOINT = true;
+    public static final boolean DEFAULT_RECOMMEND_BLOOM_FILTER = true;
+    public static final boolean DEFAULT_RECOMMEND_NO_DICTIONARY_ONHEAP_DICTIONARY_JOINT = true;
+  }
+
+  public static class PartitionRule {
+    public static final int DEFAULT_NUM_PARTITIONS = 0;
+
+    public static final long DEFAULT_THRESHOLD_MAX_SLA_PARTITION = 1000;
+    public static final long DEFAULT_THRESHOLD_MIN_QPS_PARTITION = 200;
+    public static final long DEFAULT_OPTIMAL_SIZE_PER_SEGMENT = 2000_000_000; //2GB
```

Comment:
I think 2GB is on the high end. Let's just start with 1GB probably

Reply:
changed to 1 GB